internal/backup_fetch_handler.go

```diff
@@ -44,6 +44,24 @@ func GetCommandStreamFetcher(cmd *exec.Cmd) func(folder storage.Folder, backup B
 	}
 }
 
+func GetCommandStreamFetcherParts(cmd *exec.Cmd) func(folder storage.Folder, backup Backup, fileNames []string) {
```

Comment:
The only difference from `GetCommandStreamFetcher` is one row (`downloadAndDecompressStreamParts` vs `downloadAndDecompressStream`). I suggest extracting the common code to avoid duplication.